tests/BigDecimalTest.php

```diff
@@ -163,6 +163,31 @@ public function providerOf()
         ];
     }
 
+    /**
+     * @dataProvider providerOfLocales
+     *
+     * @param string $locale
+     */
+    public function testOfValueInDifferentLocales(string $locale): void
+    {
+        setlocale(LC_ALL, $locale);
+
+        $this->assertEquals(2.5, BigDecimal::of(5/2)->toFloat());
+
+        setlocale(LC_ALL, '');
```

Comment:
Can you please save the current locale at the beginning of the method:

    $currentLocale = setlocale(LC_NUMERIC, '0');

And revert it at the end using:

    setlocale(LC_NUMERIC, $currentLocale);